storm/src/main/storm/mesos/MesosNimbus.java

```diff
@@ -249,6 +280,32 @@ public void doRegistration(final SchedulerDriver driver, Protos.FrameworkID id)
 
     _state.put(FRAMEWORK_ID, id.getValue());
     _offers = new HashMap<Protos.OfferID, Protos.Offer>();
+
+    _timer.scheduleAtFixedRate(new TimerTask() {
+      @Override
+      public void run() {
+        // performing "explicit" reconciliation; master will respond with the latest state for all logviewer tasks
+        // in the framework scheduler's statusUpdate() method
+        Collection<TaskStatus> taskStatuses = new ArrayList<TaskStatus>();
+        List<String> logviewerPaths = _zkClient.getChildren(_logviewerZkDir);
+        if (logviewerPaths == null) {
+          _driver.reconcileTasks(taskStatuses);
+          return;
+        }
+        for (String path : logviewerPaths) {
+          TaskID logviewerTaskId = TaskID.newBuilder()
+                                                .setValue(new String(_zkClient.getNodeData(String.format("%s/%s", _logviewerZkDir, path))))
```

Comment:
I think the alignment got messed up here, as well as in the `TaskStatus` creation below this.